File: python/1_Two Sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        sorted_nums = nums[:]
        sorted_nums.sort()
        i, j = 0, len(sorted_nums)-1
        while i < j:
            if sorted_nums[i] + sorted_nums[j] < target:
                i += 1
            elif sorted_nums[i] + sorted_nums[j] > target:
                j -= 1
            else:
                break
        
        for k in range(len(nums)):
            if (nums[k] == sorted_nums[i]):
                outi = k
                break
        for k in reversed(range(len(nums))):
            if (nums[k] == sorted_nums[j]):
                outj = k
                break
                
        return [outi, outj]

File: java/1_Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mp = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            int num = nums[i];
            if (mp.containsKey(target - num)) {
                return new int[]{mp.get(target - num), i};
            } else {
                mp.put(num, i);
            }
        }
        return new int[]{0, 0};
    }
}